### PR TITLE
convert Radsphere derives to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-09-27-002/radsphere-1d.html

File: Exec/radiation_tests/RadSphere/Problem_Derive.cpp [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
-    {
+    Real loc[3];
 
-        deranalytic(der, AMREX_ARLIM_ANYD(der_lo), AMREX_ARLIM_ANYD(der_hi), *nvar,
-                    data, AMREX_ARLIM_ANYD(data_lo), AMREX_ARLIM_ANYD(data_hi), *ncomp,
-                    AMREX_ARLIM_ANYD(lo), AMREX_ARLIM_ANYD(hi),
-                    AMREX_ARLIM_ANYD(domain_lo), AMREX_ARLIM_ANYD(domain_hi),
-                    AMREX_REAL_ANYD(dx), *time);
+    loc[0] = problo[0] + (static_cast<Real>(i) + 0.5_rt) * dx[0] - problem::center[0];
+    loc[1] = problo[1] + (static_cast<Real>(j) + 0.5_rt) * dx[1] - problem::center[1];
```
Comment (maxpkatz) [inline Exec/radiation_tests/RadSphere/Problem_Derive.cpp:39]: Guard against invalid accesses in 1D/2D

Comment (zingale) [reply Exec/radiation_tests/RadSphere/Problem_Derive.cpp:40]: fixed

**Comment by maxpkatz** [comment]
Can you run the tests again, just want to see if that had any effect on the diffs.

**Comment by zingale** [comment]
updated tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2020-09-28-002/radsphere-1d.html

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**